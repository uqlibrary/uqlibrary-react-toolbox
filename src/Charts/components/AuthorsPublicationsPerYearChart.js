import React from 'react';
import {PropTypes} from 'prop-types';
import Chart from './Chart';

class AuthorsPublicationsPerYearChart extends React.Component {

    static propTypes = {
        yAxisTitle: PropTypes.string,
        series: PropTypes.array,
        categories: PropTypes.array,
        className: PropTypes.string
    };

    static defaultProps = {
        yAxisTitle: 'Total publications',
        series: [],
        categories: []
    };

    constructor(props) {
        super(props);
        this.state = {
            options: {
                title: {
                    text: null
                },
                chart: {
                    type: 'column'
                },
                plotOptions: {
                    column: {
                        stacking: 'normal'
                    }
                },
                xAxis: {
                    categories: this.props.categories,
                    labels: {
                        rotation: -45,
                        y: 18
                    }
                },
                yAxis: {
                    min: 0,
                    title: {
                        text: this.props.yAxisTitle
                    },
                    stackLabels: {
                        enabled: true
                    }
                },
                legend: {
                    align: 'right',
                    verticalAlign: 'top',
                    layout: 'vertical',
                    x: -30,
                    y: -10,
                    floating: true,
                    shadow: false
                },
                tooltip: {
                    // TODO: fix formatter display of tooltip - this.x is no longer in the correct scope
                    // formatter: () => {
                    //     return '<b>' + this.x + '</b><br/>' +
                    //         this.series.name + ': ' + this.y + '<br/>' +
                    //         'Total: ' + this.point.stackTotal;
                    // }
                },
                series: this.props.series
            }
        };
    }

    render() {
        return (
            <Chart className={this.props.className + ' authors-publications-per-year-chart'} chartOptions={this.state.options} />
        );
    }
}

export default AuthorsPublicationsPerYearChart;
