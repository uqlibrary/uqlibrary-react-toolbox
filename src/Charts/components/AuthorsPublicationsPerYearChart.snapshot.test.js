/* eslint-disable */

import React from 'react';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import AuthorsPublicationsPerYearChart from './AuthorsPublicationsPerYearChart';

function setup(testProps, isShallow = false) {
    if (isShallow)
        return shallow(<AuthorsPublicationsPerYearChart {...testProps} />);

    return mount(<AuthorsPublicationsPerYearChart {...testProps} />);
}


describe('AuthorsPublicationsPerYearChart ', () => {
    it('should render empty chart component', () => {
        const app = setup({series: [], categories: [], yAxisTitle: 'title'});
        expect(toJson(app)).toMatchSnapshot();
    });
});
