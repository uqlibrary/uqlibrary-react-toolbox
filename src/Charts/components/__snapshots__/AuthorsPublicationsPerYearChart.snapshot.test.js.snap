// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsPublicationsPerYearChart  should render empty chart component 1`] = `
<AuthorsPublicationsPerYearChart
  categories={Array []}
  series={Array []}
  yAxisTitle="title"
>
  <Chart
    chartOptions={
      Object {
        "chart": Object {
          "height": 320,
          "type": "column",
        },
        "legend": Object {
          "align": "right",
          "floating": true,
          "layout": "vertical",
          "shadow": false,
          "verticalAlign": "top",
          "x": -30,
          "y": -10,
        },
        "plotOptions": Object {
          "column": Object {
            "stacking": "normal",
          },
        },
        "series": Array [],
        "title": Object {
          "text": null,
        },
        "tooltip": Object {},
        "xAxis": Object {
          "categories": Array [],
          "labels": Object {
            "rotation": -45,
            "y": 18,
          },
        },
        "yAxis": Object {
          "min": 0,
          "stackLabels": Object {
            "enabled": true,
          },
          "title": Object {
            "text": "title",
          },
        },
      }
    }
    className=" authors-publications-per-year-chart"
  >
    <div
      className=" authors-publications-per-year-chart"
    />
  </Chart>
</AuthorsPublicationsPerYearChart>
`;
